plugins/Wallet/js/locking.js

```diff
@@ -59,7 +53,9 @@ function updateStatus(result) {
 
 // Make wallet api call
 function getStatus() {
-	Siad.apiCall('/wallet', updateStatus);
+	// TODO: Don't understand why the setImmediate is needed, but without it,
+	// the '/wallet' call seems to not return right after a locking/unlocking
```

Comment:
huh. Isn't `updateStatus` supposed to only be called after the result of the api call has been returned? Are you double calling a function somewhere, or stacking events? Doesn't look like it, but I don't have a lot of ideas.